Add CIP draft: Mint Canton Coin from Unminted/Unclaimed Pool (#71)
global-synchronizer-foundation/cips: cip-00XX/cip-00XX.md

- A new action requiring confirmation (`SRARC_CreateUnallocatedUnclaimedActivityRecord`) is added to `DsoRules_ActionRequiringConfirmation`.
- A new choice `DsoRules_CreateUnallocatedUnclaimedActivityRecord` allows the DSO to propose a beneficiary, reward amount, and reason for the reward.
- Upon successful vote confirmation, an `UnallocatedUnclaimedActivityRecord` contract is created. This contract has two expiry parameters:
  - `expiresAt`: Duration from creation, derived from the configurable DSO field `unallocatedUnclaimedActivityRecordTimeout` (defaulting to 24 hours).
  - `unclaimedActivityRecordExpiresAt`: Passed to the resulting activity record once allocation occurs.

### Allocation and Expiration

- A new SV-App trigger, `UnallocatedUnclaimedActivityRecordAllocatorTrigger`, locates eligible `UnclaimedReward` contracts, archives them up to the specified amount, and creates an `UnclaimedActivityRecord`.
- A second trigger, `UnallocatedUnclaimedActivityRecordExpiryTrigger`, expires unallocated records after their TTL.
- A third trigger, `UnclaimedActivityRecordExpiryTrigger`, expires unused `UnclaimedActivityRecord` contracts.

### Minting Mechanism

> - `UnclaimedActivityRecord` is introduced as a new transferable on-ledger asset.

Reviewer: ```suggestion
- `UnclaimedActivityRecord` is introduced as a new type of activity record that can be minted.
```

  - Extend the Treasury Service to support the new activity record lifecycle.

- **Wallet App**
  - Extend the `CollectRewardsAndMergeAmuletsTrigger` to auto-collect `UnclaimedActivityRecord` using the new transfer input.

- **CLI App**
  - Introduce a Python CLI utility to calculate minted and unclaimed rewards by party ID and time range, reusing logic from `scan_txlog.py`.

- **SV App Voting UI**
  - Update the SV App voting UI to support proposals using `DsoRules_CreateUnallocatedUnclaimedActivityRecord`.

- **SV App Governance UI**
  - Add support in the SV App for configuring `unallocatedUnclaimedActivityRecordTimeout`.

## Motivation

This change is required to implement the reward distribution mechanism described in CIP-0058. 
The CIP defines a process whereby, upon successful milestone verification, the Tokenomics Working Group approves a 
reward amount and the Super Validators assign a portion of unclaimed rewards to be minted. 
This functionality leverages the standard voting process and introduces new ledger, off-ledger, and UI components to 
support reward calculation, archival of unclaimed rewards, and minting of the corresponding Amulet.

> This aligns with the first deliverable of [CIP-0058](https://github.com/global-synchronizer-foundation/cips/blob/main/cip-0058/cip-0058.md): _Enable escrow mechanisms to ensure Super Validators deliver value in return for rewards_.

Reviewer: I think it's worth mentioning that the proposed mechanism is more general and allows the SVs to mint any kind of unclaimed reward for any reason not just escrowed super validator rewards.

## Rationale

This proposal introduces a governance-backed, automated reward allocation mechanism that ensures Super Validators (SVs) never act as direct issuers of Amulets. Instead, it relies on existing voting infrastructure, explicit reward tracking, and on-ledger automation. The design satisfies the CIP-0058 objective of enabling escrowed value transfer with network approval, while preserving clear separation of duties between reward authorization and execution.

Super Validators need a way to reward parties for milestone-based work without directly issuing Amulets. To meet this goal, the system must:

- Let SVs propose reward allocations through a formal vote.
- Track and escrow those rewards in a verifiable, on-ledger structure.

> - Allow beneficiaries to receive their rewards without needing manual approval or signature from SVs.

Reviewer: not quite clear to me what you mean here. In the end the SVs do explicitly approve by creating the vote.

This design embedded additional side effects in the vote’s success path: archiving `UnclaimedReward`s and creating an `AmuletMintOffer` as part of the vote confirmation.

Rejected due to:

- Inflexibility in the current voting infrastructure, which expects one-to-one mapping between action and execution.
- Implementation would be non-trivial and hard to maintain.

#### Approach 3 – Parallel Offer and Burn Request Flow

Upon a successful vote, two contracts would be created:
1. An `AmuletMintOffer` for the beneficiary to accept.
2. A `BurnUnclaimedRewardsRequest` for the DSO to handle archival.

Rejected due to:

- Lack of atomicity, leading to potential inconsistencies.

#### Approach 4 – AmuletMintOffer Claimed Manually by Beneficiary

This design introduced `UnallocatedAmuletMintOffer`, accepted by an SV trigger that creates an `AmuletMintOffer`. The beneficiary would manually accept the final offer.

Rejected due to:

- The design would violate a core requirement of CIP-0058: SVs must not act as issuers.

### Justification for the Final Design

The chosen approach introduces a clean separation of concerns:

- Voting captures intent and approval.

> - SVs allocate funding (UnclaimedRewards) without minting.

Reviewer 2: ```suggestion
- SVs allocate funding (`UnclaimedReward` contracts) without minting.
```

- The `UnclaimedActivityRecord` encapsulates minting rights in a transferable, on-ledger structure.

> - Beneficiaries receive Amulets through a self-transfer initiated by automation in the Wallet App, maintaining signature and stakeholder integrity.

Reviewer 2: ```suggestion
- Beneficiaries receive `Amulet` through a self-transfer initiated by automation in the Wallet App, maintaining signature and stakeholder integrity.
```

Author: done
